### PR TITLE
feat(MintCollectibles): Created preview and view minted collectible pages

### LINKED ISSUE #8734: [Community Mint Collectibles]  Create new collectible preview page
### Description
[Epic](https://github.com/status-im/status-desktop/issues/8694)

Implement `Collectibles preview` page design with **mocked data**

[New collectible preview page design](https://www.figma.com/file/17fc13UBFvInrLgNUKJJg5/Kuba%E2%8E%9CDesktop?node-id=2934%3A481005&t=6E3gZk9l72B1DqRy-0)

Screenshot 2022-12-12 at 13 03 05

Related to https://github.com/status-im/status-desktop/issues/8733

### PR DESCRIPTION
**Pull request by noeliaSD**
Closes #8734 and #8737

### What does the PR do
- Created SQ 3 dots loading animation component.
- Created new mint token views: preview and created token view pages.
- Added basic navigations in mint token settings section.
- Added new related components in `storybook`.

**KNOWN ISSUES:** 
- Some integration has been done with the backend but it is a wip task that it's being developed in different steps. If you test in the app you will see there are some properties not rendered correctly (i.e. chain id and artwork).
- In general `network` related management needs to be improved in a separate task.
- Still some navigation issues that will be fixed in next task.

**SUGGESTION:** To test the UI isolated, use storybook specific pages created!

### Affected areas
 
Community Settings / Mint token

### Screenshot of functionality
**Storybook --> CommunityNewCollectibleView**

https://user-images.githubusercontent.com/97019400/223112502-cc9ce846-4c6c-4b03-9ccd-f3d7fddf4b68.mov

**Storybook --> CommunityCollectibleView**

https://user-images.githubusercontent.com/97019400/223113095-9ffa4ded-ac18-417c-ac98-fe1787c997c2.mov

**Storybook --> StatusDotsLoadingIndicator**

https://user-images.githubusercontent.com/97019400/223113271-e806111d-0842-40fd-8d5d-7f54f30f065c.mov

**App flow (wip - Some backend integration already pending):**

https://user-images.githubusercontent.com/97019400/223117013-cba12b90-8607-401c-90ad-06e066ee13f6.mov

### CODE REVIEW

File: ui/app/AppLayouts/Chat/panels/communities/TokenHoldersPanel.qml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        anchors.fill: parent
+        spacing: Style.current.padding
+
+        StatusBaseText {
+            visible: !root.preview
+            Layout.fillWidth: true
+            wrapMode: Text.Wrap
+            font.pixelSize: Style.current.primaryTextFontSize
+            color: Theme.palette.baseColor1
+            text: qsTr("All Anniversary token holders")
```
[the remark this answers is not in the document]
Comment (noeliaSD) [reply ui/app/AppLayouts/Chat/panels/communities/TokenHoldersPanel.qml:37]: Ups!! Will update it! :)

File: ui/app/AppLayouts/Chat/panels/communities/MintTokensFooterPanel.qml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                icon.name: "token-sale"
+                text: qsTr("Retail")
+
+                onClicked: root.retailClicked()
+            }
+
+            StatusFlatButton {
+                id: remotelySelfDestructButton
+
+                icon.name: "airdrop" // TO UPDATE!!
```
[the remark this answers is not in the document]
Comment (noeliaSD) [reply ui/app/AppLayouts/Chat/panels/communities/MintTokensFooterPanel.qml:59]: Yesss!! Will update as well!! thanks!!

File: storybook/pages/CommunityMintTokensSettingsPanelPage.qml [hunk 3]
```diff
@@ -29,7 +29,7 @@ SplitView {
                 }
             }
             communitiesStore: QtObject {
-
+                readonly property var holdersModel: TokenHoldersModel {}
```
Comment (alexjba) [inline storybook/pages/CommunityMintTokensSettingsPanelPage.qml:32]:
```suggestion
                readonly property TokenHoldersModel holdersModel: TokenHoldersModel {}
```

**Review by alexjba — APPROVED** [review]
Very nice work! 👍 
One question: Shouldn't the network selector close after selection? Usually such menus with single selection will close after selection.

https://user-images.githubusercontent.com/47811206/223156606-823647a1-d789-455f-a74b-268befa9ed1b.mov

**Comment by noeliaSD** [comment]
> Very nice work! 👍 One question: Shouldn't the network selector close after selection? Usually such menus with single selection will close after selection.

Ok, I will look at the network improvements in my next pr. Thanks for pointing this out. 
Since this component is a reusable one from wallet and comes from a multiselection component, I think that this is the reason it is not closed after one item is selected!

File: ui/app/AppLayouts/Chat/views/communities/CommunityNewCollectibleView.qml [hunk 4]
```diff
@@ -15,37 +15,35 @@ StatusScrollView {
...
-    property var tokensModel // TEMPORARY
     property int viewWidth: 560 // by design
 
+    // Collectible properties
+    property alias name: nameInput.text
+    property alias symbol: symbolInput.text
+    property alias description: descriptionInput.text
+    property alias supplyText: supplyInput.text
+    property alias infiniteSupply: unlimitedSupplyChecker.checked
+    property alias transferable: transferibleChecker.checked
```
Comment (micieslak) [inline ui/app/AppLayouts/Chat/views/communities/CommunityNewCollectibleView.qml:26]: transferibleChecker -> transferableChecker

same in checkbox label :)

**Review by micieslak — CHANGES_REQUESTED** [review]
Nice work! Some minor problems only.

1. When network is "untouched", the preview displays empty item:
Screenshot from 2023-03-07 11-06-57
and warning is emited:
```
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Chat/views/communities/CommunityCollectibleView.qml:183:21: QML SVGImage: Cannot open: file:///home/michalc/workspace/status-desktop/ui/imports/assets/icons/.svg
```
2. Total supply number seem to be not localized:
Screenshot from 2023-03-07 11-06-08

Also localized input is not accepted:

Screenshot from 2023-03-07 11-11-25

3. Probably symbol name should be aligned to the left side as well (not centered):
Screenshot from 2023-03-07 11-06-57


4. When creating second collectible, inputs are red from the start:
Screenshot from 2023-03-07 11-07-05

5. Tiny detail - icon in the combobox on the top is not anti-aliased:
Screenshot from 2023-03-07 09-56-08

6. Token symbol should be somehow limited as well imo:
Screenshot from 2023-03-07 09-54-45

7. diacritic are not supported well (not sure if we want to ban them, and now they are countes as 2)
Screenshot from 2023-03-07 09-54-14

File: storybook/pages/CommunityNewCollectibleViewPage.qml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+        Item {
+            SplitView.fillWidth: true
+            SplitView.fillHeight: true
+
+            CommunityNewCollectibleView {
+                anchors.fill: parent
+                anchors.margins: 50
+                store: QtObject {
+                    property var layer1Networks: NetworksModel.layer1Networks
```
Comment (micieslak) [inline storybook/pages/CommunityNewCollectibleViewPage.qml:31]: is it better to have so many models for that? What about one model with some category role or sth?

Comment (noeliaSD) [reply storybook/pages/CommunityNewCollectibleViewPage.qml:31]: Will keep it as it is following the backend structure we have right now!

**Comment by noeliaSD** [comment]
> Nice work! Some minor problems only.
> 
> 1. When network is "untouched", the preview displays empty item:
>    Screenshot from 2023-03-07 11-06-57
>    and warning is emited:
> 
> ```
> file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Chat/views/communities/CommunityCollectibleView.qml:183:21: QML SVGImage: Cannot open: file:///home/michalc/workspace/status-desktop/ui/imports/assets/icons/.svg
> ```
> 
> 2. Total supply number seem to be not localized:
>    Screenshot from 2023-03-07 11-06-08
> 
> Also localized input is not accepted:
> 
> Screenshot from 2023-03-07 11-11-25
> 
> 3. Probably symbol name should be aligned to the left side as well (not centered):
>    Screenshot from 2023-03-07 11-06-57
> 4. When creating second collectible, inputs are red from the start:
>    Screenshot from 2023-03-07 11-07-05
> 5. Tiny detail - icon in the combobox on the top is not anti-aliased:
>    Screenshot from 2023-03-07 09-56-08
> 6. Token symbol should be somehow limited as well imo:
>    Screenshot from 2023-03-07 09-54-45
> 7. diacritic are not supported well (not sure if we want to ban them, and now they are countes as 2)
>    Screenshot from 2023-03-07 09-54-14

Thanks for pointing everything out. I will fix some of the issues in my next pr.

**[1 and 5]** As told before, network management will be improved later.
**[6 and 7]** I will ping the design team to ensure the validation critera for all fields is correct and will introduce the rules in the next pr.
**[3]** The design is explicitly like this but I will also check it with designers and will update in next pr if needed.
**[4]**  I will improve data management in next pr and will solve that.
**[2]** Ticked created to track this issue: https://github.com/status-im/status-desktop/issues/9763

**Review by micieslak — APPROVED** [review]
LGTM!

**Pull request merged**